jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/conf/JdbcStorageConfigManager.java

```diff
@@ -28,28 +37,38 @@
  */
 public class JdbcStorageConfigManager {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcStorageConfigManager.class);
   public static final String CONFIG_PREFIX = "hive.sql";
+  public static final String CONFIG_PWD = CONFIG_PREFIX + ".dbcp.password";
   private static final EnumSet<JdbcStorageConfig> DEFAULT_REQUIRED_PROPERTIES =
     EnumSet.of(JdbcStorageConfig.DATABASE_TYPE,
-        JdbcStorageConfig.JDBC_URL,
-        JdbcStorageConfig.JDBC_DRIVER_CLASS,
         JdbcStorageConfig.QUERY);
 
-
   private JdbcStorageConfigManager() {
   }
 
-
   public static void copyConfigurationToJob(Properties props, Map<String, String> jobProps) {
     checkRequiredPropertiesAreDefined(props);
+    resolveMetadata(props);
     for (Entry<Object, Object> entry : props.entrySet()) {
-      jobProps.put(String.valueOf(entry.getKey()), String.valueOf(entry.getValue()));
+      if (!String.valueOf(entry.getKey()).equals(CONFIG_PWD)) {
+        jobProps.put(String.valueOf(entry.getKey()), String.valueOf(entry.getValue()));
+      }
     }
   }
 
+  public static void copySecretsToJob(Properties props, Map<String, String> jobSecrets) {
+    checkRequiredPropertiesAreDefined(props);
+    resolveMetadata(props);
+    String secret = props.getProperty(CONFIG_PWD);
+    if (secret != null) {
+      jobSecrets.put(CONFIG_PWD, secret);
```

Comment:
This will be retried and then transferred into the credentials object. But yes, at this point it's in clear text in the code. I don't think this is a problem though.